AdClickthroughWorkerTests.swift

```diff
@@ -0,0 +1,128 @@
+//  Copyright 2019, Oath Inc.
+//  Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
+
+import XCTest
+import PlayerCore
+import SafariServices
+@testable import VerizonVideoPartnerSDK
+
+class AdClickthroughWorkerTests: XCTestCase {
+    let recorder = Recorder()
+    
+    let testURL = URL(string: "http://")!
+    let controller = SFSafariViewController(url: URL(string: "http://")!)
+    let mp4Creative = AdCreative.MP4(url: URL(string: "http://")!,
+                                     clickthrough: URL(string: "http://")!,
+                                     pixels: AdPixels(),
+                                     id: nil,
+                                     width: 10,
+                                     height: 10,
+                                     scalable: false,
+                                     maintainAspectRatio: true)
+    let vpaidCreative = AdCreative.VPAID(url: URL(string: "http://")!,
+                                         adParameters: "",
+                                         clickthrough: URL(string: "http://")!,
+                                         pixels: AdPixels(),
+                                         id: "")
+    let vpaidCreativeWithoutCllick = AdCreative.VPAID(url: URL(string: "http://")!,
+                                                      adParameters: "",
+                                                      clickthrough: nil,
+                                                      pixels: AdPixels(),
+                                                      id: "")
+    
+    var sut: PlayerViewController.AdClickthroughWorker!
+    
+    override func setUp() {
+        let showSafariHook = recorder.hook("showSafariHook")
+        sut = PlayerViewController.AdClickthroughWorker(
+            showSafari: { _, _ in
+                showSafariHook()
+        },
+            safariFinishHandler: recorder.hook("finishSafariHook") { rcd, cmp -> Bool in
```

Comment:
`safariFinishHandler: recorder.hook("finishSafariHook")` - this doesn't work?

Reply:
It doesn't.

Reply:
It works - just checked it.

Reply:
Got it. I had compile errors, dunno why =\